### PR TITLE
Feature/more validators

### LINKED ISSUE #32: add more sophisticated validators

### PR DESCRIPTION
**Pull request by buehler**
Fixes #32 
- adds parameterized validator factories for isString and isNumber
- both have options now (object destructuring)

### CODE REVIEW

**Comment by buehler** [comment]
Do we need more validators or are these provided enough?

File: validators/Validators.spec.ts [hunk 1]
```diff
@@ -11,50 +11,111 @@ describe('Validators', () => {
...
+            isString()(null).should.be.false;
         });
 
         it('should validate an undefined value correctly', () => {
-            isStringValidator(undefined).should.be.false;
+            isString()(undefined).should.be.false;
+        });
+
+        it('should validator min length correctly', () => {
+            isString({minLength: 3})('aaa').should.be.true;
```
Comment (dwirz) [inline validators/Validators.spec.ts]: Why not min/max?

File: validators/Validators.ts [hunk 2]
```diff
@@ -1,19 +1,89 @@
...
+ * @param {number} [maxLength] - Maximum length of the string.
+ * @returns {Validator} - Validator function for the given parameters.
+ */
+export function isString({allowEmpty = false, minLength, maxLength}: {allowEmpty?: boolean, minLength?: number, maxLength?: number} = {}): Validator {
+    return (value: string) => {
+        if (isNullOrUndefined(value) || typeof value !== 'string') {
+            return false;
+        }
+
+        if ((!allowEmpty && value === '')) {
```
Comment (dwirz) [inline validators/Validators.ts]: Too many brackets... ;)

Comment (buehler) [inline validators/Validators.ts]: e676d5b

**Pull request merged**